### PR TITLE
Re-organize README to clarify for open source contributors

### PR DESCRIPTION
**Pull request by fukawi2**
### Details

Reorganizing the README file to make it clearer which parts apply to open source contributors, and which is for Expensify employees.

### Fixed Issues
https://expensify.slack.com/archives/C01GTK53T8Q/p1608692346084200

### Tests
1. Follow the instructions and verify that each platform can be set up successfully

### Screenshots
N/A

### CODE REVIEW

**Comment by fukawi2** [comment]
I have read the CLA Document and I hereby sign the CLA

File: README.md [hunk 1]
```diff
@@ -123,32 +107,34 @@ You are first required to build the tests, then you can run them:
 1. To build the **Detox end to end tests**: `npm run detox-build`
 2. To run the **Detox end to end tests**: `npm run detox-test`
 
-## Troubleshooting
+# Troubleshooting
 1. If you are having issues with **_Getting Started_**, please reference [React Native's Documentation](https://reactnative.dev/docs/environment-setup)
-2. If you are running into issues communicating with `expensify.com.dev` (CORS, SSL, etc.), running via `ngrok` is recommended, see step 3 in **_Getting Started_**
+2. If you are running into issues communicating with the API please verify your `.env` file is set up correctly for the platform you are trying to run.
```
Comment (timszot) [inline README.md:112]: NAB: Can we link to a header in the README that takes them back to `## Getting started` section that explains all this?

**Review by timszot — COMMENTED** [review]
LGTM, left one NAB comment. Would like some more eyes on this overall that are more familiar with the setup to see if they can think of any other improvements.

**Review by puneetlath — DISMISSED** [review]
Looks great! Much clearer.

File: README.md (hunk 1 above)
Comment (marcaaron) [reply README.md:112]: Sure! But to clarify what specifically are you wanting to be added to the Getting Started section?

Comment (timszot) [reply README.md:112]: Oh, yeah, that could have been clearer lol

I'm thinking we add a link on `set up correctly` that links back to the `Getting Started` section of this doc.

Comment (timszot) [reply README.md:112]: I see your newest commit, and that works, but yeah, I was more thinking right in the middle of where it talks about what to do if you're having problems, here's a link of where to read again.

**Review by timszot — DISMISSED** [review]
Cool! No other comments from me.

**Comment by fukawi2** [comment]
Can we add a comment to `.env.production` explaining that those credentials are supposed to be there and we haven't accidentally committed sensitive data to GH?

**Comment by timszot** [comment]
> Can we add a comment to .env.production explaining that those credentials are supposed to be there and we haven't accidentally committed sensitive data to GH?

I think that's a good idea, would you like to add that @fukawi2 since you've already contributed to this PR as well?

**Comment by fukawi2** [comment]
Done! :)

**Review by timszot — APPROVED** [review]

**Review by puneetlath — APPROVED** [review]

**Review by timszot — APPROVED** [review]

**Pull request merged**